chainerio/__init__.py

```diff
@@ -68,60 +87,194 @@ def open(file_path: str, mode: str = 'rb',
+
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     default_context.set_root(uri_or_handler)
 
 
-def create_handler(uri: str) -> IO:
+def create_handler(handler_name: str) -> IO:
+    """Returns a handler according to the given ``handler_name``
+
+    The current supported handlers are:
+
+    1. posix
+    2. hdfs
+
+    Args:
+        handler_name (str): the name of handler to create
+
+    Returns:
+        An object of :class:`chainerio.IO`
+        according to the ``handler_name``
+
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     (handler, actual_path, is_URI) = \
-        default_context.get_handler_by_name(uri)
+        default_context.get_handler_by_name(handler_name)
     return handler
 
 
-def isdir(file_path: str) -> bool:
+def isdir(path: str) -> bool:
+    """Returns ``Ture`` if the path is an existing directory
+
+    Forward to the corresponding :func:`IO.isdir` upon the default handler.
+    Refer to the link for more details.
+
+    The ``path`` can be  an Unix path or an URI.
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     (handler, actual_path, is_URI) = \
-        default_context.get_handler_by_name(file_path)
+        default_context.get_handler_by_name(path)
     return handler.isdir(actual_path)
 
 
-def mkdir(file_path: str, mode: int = 0o777, **kwargs) -> None:
+def mkdir(path: str, mode: int = 0o777, *,
+          dir_fd: Optional[int] = None) -> None:
+    """Makes a directory with mode
+
+    Forward to the corresponding :func:`IO.makedir` upon the default handler.
+    Refer to the link for more details.
+
+    The ``path`` can be  an Unix path or an URI.
+
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     (handler, actual_path, is_URI) = \
-        default_context.get_handler_by_name(file_path)
-    return handler.mkdir(actual_path, mode, *kwargs)
+        default_context.get_handler_by_name(path)
+    return handler.mkdir(actual_path, mode, dir_fd=dir_fd)
 
 
-def makedirs(file_path: str, mode: int = 0o777,
+def makedirs(path: str, mode: int = 0o777,
              exist_ok: bool = False) -> None:
+    """Makes directories recursively with mode
+
+    Forward to the corresponding :func:`IO.makedirs` upon the default handler.
+    Refer to the link for more details.
+
+    The ``path`` can be  an Unix path or an URI.
+
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     (handler, actual_path, is_URI) = \
-        default_context.get_handler_by_name(file_path)
+        default_context.get_handler_by_name(path)
     return handler.makedirs(actual_path, mode, exist_ok)
 
 
-def exists(file_path: str) -> bool:
+def exists(path: str) -> bool:
+    """Returns ``True`` when the given ``path`` exists
+
+    Forward to the corresponding :func:`IO.exists` upon the default handler.
+    Refer to the link for more details.
```

Comment:
Ditto. But what happens with symlinks?

Reply:
It returns ``False`` when the original file is not exist, just like the `os.path.exists`. No problem for HDFS, where symblink is not supported